libs/sitemaps/src/createDatabasePaginator.ts

```diff
@@ -134,6 +134,7 @@ export function createDatabasePaginatorWithAdapter(
 
   let records: ReadonlyArray<Link> = [];
 
+  // eslint-disable-next-line @typescript-eslint/require-await
   async function hasNext() {
```

Comment:
It won't match the interface then...

Reply:
So we should update the interface since neither implementation of Paginator.hasNext uses promises.

Reply:
Only because of the implementation. It's using the iterator pattern. In this case async iterator. Both hasNext and next() need to use promises because then we can't change the implementation without rewriting all the code that uses it and then we can never change the underlying implementation. 

I mean I might be overthinking a bit because this is just for the sitemaps but I was thinking we could use this for anything that needs postgres pagination.